File: 2da Forma/GlobalResponse.java
// Framework: Spring Boot
// Archivo: GlobalResponse.java

package com.example.common.responses;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class GlobalResponse<T> {
    
    private Boolean success;
    private String message;
    private T data;
    private Object errors;
    private PaginationInfo pagination;

    // Success responses
    public static <T> GlobalResponse<T> ok(String message) {
        return GlobalResponse.<T>builder()
                .success(true)
                .message(message != null ? message : "Operación exitosa")
                .build();
    }

    public static <T> GlobalResponse<T> ok() {
        return ok("Operación exitosa");
    }

    public static <T> GlobalResponse<T> okWithData(T data, String message) {
        return GlobalResponse.<T>builder()
                .success(true)
                .message(message != null ? message : "Operación exitosa")
                .data(data)
                .build();
    }

    public static <T> GlobalResponse<T> okWithData(T data) {
        return okWithData(data, "Operación exitosa");
    }

    public static <T> GlobalResponse<T> okPaginated(
            T data,
            int totalItems,
            int currentPage,
            int pageSize,
            String message
    ) {
        PaginationInfo pagination = PaginationInfo.builder()
                .totalItems(totalItems)
                .currentPage(currentPage)
                .pageSize(pageSize)
                .totalPages((int) Math.ceil((double) totalItems / pageSize))
                .hasPrevious(currentPage > 1)
                .hasNext(currentPage < Math.ceil((double) totalItems / pageSize))
                .build();

        return GlobalResponse.<T>builder()
                .success(true)
                .message(message != null ? message : "Datos obtenidos exitosamente")
                .data(data)
                .pagination(pagination)
                .build();
    }

    public static <T> GlobalResponse<T> okPaginated(
            T data,
            int totalItems,
            int currentPage,
            int pageSize
    ) {
        return okPaginated(data, totalItems, currentPage, pageSize, "Datos obtenidos exitosamente");
    }

    // Error responses
    public static <T> GlobalResponse<T> fail(String message) {
        return GlobalResponse.<T>builder()
                .success(false)
                .message(message)
                .build();
    }

    public static <T> GlobalResponse<T> failWithErrors(String message, Object errors) {
        return GlobalResponse.<T>builder()
                .success(false)
                .message(message)
                .errors(errors)
                .build();
    }

    public static <T> GlobalResponse<T> validationError(
            String message,
            Map<String, List<String>> validationErrors
    ) {
        return GlobalResponse.<T>builder()
                .success(false)
                .message(message)
                .errors(validationErrors)
                .build();
    }

    public static <T> GlobalResponse<T> notFound(String message) {
        return GlobalResponse.<T>builder()
                .success(false)
                .message(message != null ? message : "Recurso no encontrado")
                .build();
    }

    public static <T> GlobalResponse<T> notFound() {
        return notFound("Recurso no encontrado");
    }

    public static <T> GlobalResponse<T> unauthorized(String message) {
        return GlobalResponse.<T>builder()
                .success(false)
                .message(message != null ? message : "No autorizado")
                .build();
    }

    public static <T> GlobalResponse<T> unauthorized() {
        return unauthorized("No autorizado");
    }

    public static <T> GlobalResponse<T> serverError(String message) {
        return GlobalResponse.<T>builder()
                .success(false)
                .message(message != null ? message : "Error interno del servidor")
                .build();
    }

    public static <T> GlobalResponse<T> serverError() {
        return serverError("Error interno del servidor");
    }

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class PaginationInfo {
        private Integer totalItems;
        private Integer currentPage;
        private Integer pageSize;
        private Integer totalPages;
        private Boolean hasPrevious;
        private Boolean hasNext;
    }
} {
    
}

File: 2da Forma/global-response.py
# Framework: Flask
# Archivo: global_response.py

from typing import TypeVar, Generic, Optional, Any, Dict, List
from dataclasses import dataclass, asdict, field
from math import ceil

T = TypeVar('T')

@dataclass
class PaginationInfo:
    total_items: int
    current_page: int
    page_size: int
    total_pages: int = field(init=False)
    has_previous: bool = field(init=False)
    has_next: bool = field(init=False)

    def __post_init__(self):
        self.total_pages = ceil(self.total_items / self.page_size) if self.page_size > 0 else 0
        self.has_previous = self.current_page > 1
        self.has_next = self.current_page < self.total_pages

    def to_dict(self) -> dict:
        return asdict(self)

@dataclass
class GlobalResponse(Generic[T]):
    success: bool
    message: str
    data: Optional[T] = None
    errors: Optional[Any] = None
    pagination: Optional[PaginationInfo] = None

    def to_dict(self) -> dict:
        result = {
            'success': self.success,
            'message': self.message
        }
        if self.data is not None:
            result['data'] = self.data
        if self.errors is not None:
            result['errors'] = self.errors
        if self.pagination is not None:
            result['pagination'] = self.pagination.to_dict()
        return result

    # Success responses
    @classmethod
    def ok(cls, message: str = "Operación exitosa") -> "GlobalResponse[T]":
        return cls(success=True, message=message)

    @classmethod
    def ok_with_data(cls, data: T, message: str = "Operación exitosa") -> "GlobalResponse[T]":
        return cls(success=True, message=message, data=data)

    @classmethod
    def ok_paginated(
        cls,
        data: T,
        total_items: int,
        current_page: int,
        page_size: int,
        message: str = "Datos obtenidos exitosamente"
    ) -> "GlobalResponse[T]":
        pagination = PaginationInfo(
            total_items=total_items,
            current_page=current_page,
            page_size=page_size
        )
        return cls(success=True, message=message, data=data, pagination=pagination)

    # Error responses
    @classmethod
    def fail(cls, message: str) -> "GlobalResponse[T]":
        return cls(success=False, message=message)

    @classmethod
    def fail_with_errors(cls, message: str, errors: Any) -> "GlobalResponse[T]":
        return cls(success=False, message=message, errors=errors)

    @classmethod
    def validation_error(
        cls,
        message: str,
        validation_errors: Dict[str, List[str]]
    ) -> "GlobalResponse[T]":
        return cls(success=False, message=message, errors=validation_errors)

    @classmethod
    def not_found(cls, message: str = "Recurso no encontrado") -> "GlobalResponse[T]":
        return cls(success=False, message=message)

    @classmethod
    def unauthorized(cls, message: str = "No autorizado") -> "GlobalResponse[T]":
        return cls(success=False, message=message)

    @classmethod
    def server_error(cls, message: str = "Error interno del servidor") -> "GlobalResponse[T]":
        return cls(success=False, message=message)